frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/IActionSourceAwareBranchNameProvider.java

```diff
@@ -0,0 +1,14 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import io.vavr.control.Option;
+
+public interface IActionSourceAwareBranchNameProvider {
+  enum ActionSource {
+    CONTEXTMENU, TOOLBAR
+  }
+
+  ActionSource getActionSource();
```

Comment:
As you mentioned yourself on Slack, consider getting this value of `AnActionEvent` instead